api/handlers.go

```diff
@@ -412,17 +456,28 @@ func (api *API) userDELETE(w http.ResponseWriter, req *http.Request, _ httproute
 
 // userPOST creates a new user.
 func (api *API) userPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	email := req.PostFormValue("email")
+	defer api.closeBody(req)
+	// Parse the request's body.
+	var payload credentialsDTO
+	b, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
+		return
+	}
+	err = json.Unmarshal(b, &payload)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "failed to parse request body"), http.StatusBadRequest)
```

Comment:
update error message

Reply:
This one is fine. But the one at 464 should say "failed to read request body". Fixed.